gcsio/src/test/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageTest.java

```diff
@@ -3387,6 +3397,48 @@ public void testInstantiateAuthorizationHandler() {
     assertThat(authorizer).isNotNull();
   }
 
+  @Test
+  public void initializeRequest_withAccessTokenProviderNotUsingNewTokenPerRequest()
+      throws IOException {
+    MockHttpTransport transport = mockTransport(jsonDataResponse(newBucket(BUCKET_NAME)));
+
+    GoogleCloudStorageOptions gcsOptions = GCS_OPTIONS.toBuilder()
+        .setRefreshAccessTokenPerRequestEnabled(false)
+        .build();
+
+    StorageAccessTokenProvider mockAtp = Mockito.mock(StorageAccessTokenProvider.class);
```

Comment:
Use stub/fake instead, we try to not use Mockito mocks if possible

Reply:
understood, I will keep this one since it's very straight forward and I'm looking at `verify` functionality.

Reply:
While it's straightforward now, if we do not have stub for `StorageAccessTokenProvider` it means that when we start to work on new functionality we will need to implement it and will end up with mocks again, that's why I think that better to do it right away and use it here.

Reply:
I will add Fake/stub. But I'm even more confused now. If we add more functionality, we will have to update the stub/fake. no? what am I missing?